src/main/java/appland/notifications/AppMapNotifications.java

```diff
@@ -87,4 +90,29 @@ public static void showExpandedRecordingNotification(@NotNull Project project,
             notification.notify(project);
         }, ModalityState.any());
     }
+
+    public static void showTelemetryNotification(@NotNull Project project,
+    @Nullable String title,
+    @NotNull String content,
+    @NotNull NotificationType type,
+    @NotNull Consumer<Boolean> onDismiss) {
+        ApplicationManager.getApplication().invokeLater(() -> {
+            Notification notification = new AppMapFullContentNotification(
+                        TELEMETRY_ID, null,
+                        title, null, content,
+                        type, null
+                );
+
+            var denyAction = NotificationAction.create(lazy("telemetry.permission.deny"), (e, n) -> {
+                onDismiss.accept(false);
+                n.expire();
+            });
+
+            notification = notification.addAction(denyAction);
+            notification.whenExpired(() -> {
+                onDismiss.accept(true);
```

Comment:
This is always enabling telemetry, even after the user clicked "Opt out".
A click on "Opt out" first sets it to disabled and then expires the notification. The expiry triggers this `whenExpired`, which resets it back to default.

Reply:
Thanks, I'm just going to drop this statement.